Sources/GRPCCore/GRPCClient.swift

```diff
@@ -112,13 +112,18 @@ public final class GRPCClient: Sendable {
   /// The transport which provides a bidirectional communication channel with the server.
   private let transport: any ClientTransport
 
-  /// A collection of interceptors providing cross-cutting functionality to each accepted RPC.
+  private let interceptorPipeline: [ClientInterceptorPipelineOperation]
+
+  /// A collection of interceptors providing cross-cutting functionality to each accepted RPC, keyed by the method to which they apply.
+  ///
+  /// The list of interceptors for each method is computed from `interceptorsPipeline` when calling a method for the first time.
+  /// This caching is done to avoid having to compute the applicable interceptors for each request made.
   ///
   /// The order in which interceptors are added reflects the order in which they are called. The
   /// first interceptor added will be the first interceptor to intercept each request. The last
   /// interceptor added will be the final interceptor to intercept each request before calling
   /// the appropriate handler.
-  private let interceptors: [any ClientInterceptor]
+  private let interceptorsPerMethod: Mutex<[MethodDescriptor: [any ClientInterceptor]]>
```

Comment:
I thought this caching could be useful to speed things up, similar to what we do in the `RPCRouter` on the server. However I wonder if we care more about the client not taking up too much space (which could be the case if it calls _many_ methods and this dictionary grows a lot).